### PR TITLE
Add autoround xpu ut

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

UT

## Description

for BMG
pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/xpu
If encountering ipex error: pip uninstall intel_extension_for_pytorch
or use nightly
pip3 install --pre torch torchvision torchaudio --index-url <a href="https://download.pytorch.org/whl/nightly/xpu" rel="noreferrer noopener" title="https://download.pytorch.org/whl/nightly/xpu" target="_blank">https://download.pytorch.org/whl/nightly/xpu</a>
or
pip3 install torch==2.8 torchvision torchaudio --index-url https://download.pytorch.org/whl/xpu

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

___

### **PR Type**
Tests

___

### **Description**
- Add XPU unit tests for autoround

- Test various schemes and formats

- Include VLM and LM head quantization tests

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["Add XPU UT"] -- "Test schemes" --> B["Test formats"]
  B -- "Test VLM models" --> C["Test LM head quantization"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>test_autoround.py</strong><dd><code>Add XPU autoround unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

test/3x/torch/quantization/weight_only/test_autoround.py

<ul><li>Added XPU unit tests for autoround<br> <li> Included tests for different schemes and formats<br> <li> Added tests for VLM models and LM head quantization</ul>

</details>

  </td>
  <td><a href="https://github.com/intel/neural-compressor/pull/2332/files#diff-5767d835ec25fe413a700db09e70c11fb5bb483735cffd7aa958d96cf8c9c377">+161/-0</a>&nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

**Comment by PRAgent4INC** [comment]
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 3 🔵🔵🔵⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>PR contains tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/intel/neural-compressor/pull/2332/files#diff-5767d835ec25fe413a700db09e70c11fb5bb483735cffd7aa958d96cf8c9c377R1-R100'><strong>Possible Issue</strong></a>

Ensure that all test cases cover edge cases and invalid inputs to prevent unexpected behavior.
</summary>

```python
import copy
import shutil

import pytest
import torch
import transformers
from packaging.version import Version, parse
import os
from functools import lru_cache

@lru_cache(None)
def is_habana_framework_installed():
    """Check if Habana framework is installed.

    Only check for the habana_frameworks package without importing it to avoid
    initializing lazy-mode-related components.
    """
    from importlib.util import find_spec

    package_spec = find_spec("habana_frameworks")
    return package_spec is not None

def set_hpu_torch_compile_envs():
    if not is_habana_framework_installed():
        return None
    import torch._dynamo.config as dynamo_config
    import torch._inductor.config as inductor_config

    os.environ["PT_HPU_LAZY_MODE"] = "0"
    os.environ["PT_ENABLE_INT64_SUPPORT"] = "1"
    inductor_config.force_disable_caches = True
    dynamo_config.inline_inbuilt_nn_modules = True


# The `TestAutoRoundHPU` is expected to be run with `compile` mode,
# so set the HPU environment variables before importing INC.
if is_habana_framework_installed():
    set_hpu_torch_compile_envs()


from neural_compressor.torch.quantization import (
    AutoRoundConfig,
    convert,
    get_default_AutoRound_config,
    prepare,
    quantize,
)

from neural_compressor.torch.utils import logger

torch.backends.__allow_nonbracketed_mutation_flag = True

try:
    import auto_round
    from auto_round.export.export_to_itrex.model_wrapper import WeightOnlyLinear

    auto_round_installed = True
except ImportError:
    auto_round_installed = False

try:
    import compressed_tensors

    ct_installed = True
except ImportError:
    ct_installed = False


@torch.no_grad()
def run_fn(model, dataloader):
    for data in dataloader:
        if isinstance(data, tuple) or isinstance(data, list):
            model(*data)
        elif isinstance(data, dict):
            model(**data)
        else:
            model(data)

@pytest.mark.skipif(is_habana_framework_installed(), reason="These tests are not supported on HPU for now.")
@pytest.mark.skipif(not auto_round_installed, reason="auto_round module is not installed")
class TestAutoRoundCPU:
    @classmethod
    def setup_class(self):
        self.gptj = transformers.AutoModelForCausalLM.from_pretrained(
            "hf-internal-testing/tiny-random-GPTJForCausalLM",
            torchscript=True,
        )
        self.inp = torch.ones([1, 10], dtype=torch.long)
        self.tokenizer = transformers.AutoTokenizer.from_pretrained(
            "hf-internal-testing/tiny-random-GPTJForCausalLM", trust_remote_code=True
        )
        from neural_compressor.torch.algorithms.weight_only.autoround import get_dataloader
        self.dataloader = get_dataloader(self.tokenizer, 32, dataset_name="NeelNanda/pile-10k", seed=42, bs=8, nsamples=10)
        self.label = self.gptj(self.inp)[0]

    @classmethod
    def teardown_class(self):
        shutil.rmtree("saved_results", ignore_errors=True)

    def setup_method(self, method):
```

</details>

</td></tr>
</table>

**Comment by PRAgent4INC** [comment]
## PR Code Suggestions ✨

File: test/3x/torch/quantization/weight_only/test_autoround.py [hunk 1]
```diff
@@ -571,3 +571,164 @@ def test_autoround_with_quantize_API(self):
             run_args=(self.dataloader,),
         )
         assert isinstance(q_model.model.layers[0].self_attn.k_proj, WeightOnlyLinear), "packing model failed."
+
+@pytest.mark.skipif(is_xpu_available=True, reason="These tests are not supported on XPU for now.")
```
Comment (chensuyue) [inline test/3x/torch/quantization/weight_only/test_autoround.py]: only skip when no xpu detected.

**Review by chensuyue — APPROVED** [review]
